core/src/main/java/org/infinispan/distribution/group/GroupManagerImpl.java

```diff
@@ -46,13 +48,16 @@ public GroupMetadataImpl(Method method) {
 
         @Override
         public String getGroup(Object instance) {
-            return String.class.cast(invokeAccessibly(instance, method, Util.EMPTY_OBJECT_ARRAY));
+            PrivilegedAction<Object> action = () -> invokeAccessibly(instance, method, Util.EMPTY_OBJECT_ARRAY);
+            Object object = System.getSecurityManager() == null ? action.run() : AccessController.doPrivileged(action);
+            return String.class.cast(object);
```

Comment:
Instead of always invoking the lambda like that I'd rather do
if (System.getSecurityManager() == null) {
return String.class.cast(invokeAccessibly(instance, method, Util.EMPTY_OBJECT_ARRAY));
} else {
... // priv action
}